src/dhcpmon/src/dhcp_device.cpp

```diff
@@ -163,86 +213,102 @@ static void handle_dhcp_option_53(dhcp_device_context_t *context,
 }
 
 /**
- * @code read_callback(fd, event, arg);
+ * @code client_packet_handler(dhcp_device_context_t *context, ssize_t buffer_sz);
+ *
+ * @brief packet handler to process received rx and tx packets
+ *
+ * @param context       pointer to device (interface) context
+ * @param buffer_sz     buffer that stores received packet data
+ *
+ * @return none
+ */
+static void client_packet_handler(dhcp_device_context_t *context, ssize_t buffer_sz)
+{
+    struct ether_header *ethhdr = (struct ether_header*) context->buffer;
+    struct ip *iphdr = (struct ip*) (context->buffer + IP_START_OFFSET);
+    struct udphdr *udp = (struct udphdr*) (context->buffer + UDP_START_OFFSET);
+    uint8_t *dhcphdr = context->buffer + DHCP_START_OFFSET;
+    int dhcp_option_offset = DHCP_START_OFFSET + DHCP_OPTIONS_HEADER_SIZE;
+
+    if (((unsigned)buffer_sz > UDP_START_OFFSET + sizeof(struct udphdr) + DHCP_OPTIONS_HEADER_SIZE) &&
+        (ntohs(udp->len) > DHCP_OPTIONS_HEADER_SIZE))
+    {
+        int dhcp_sz = ntohs(udp->len) < buffer_sz - UDP_START_OFFSET - sizeof(struct udphdr) ?
+                    ntohs(udp->len) : buffer_sz - UDP_START_OFFSET - sizeof(struct udphdr);
+        int dhcp_option_sz = dhcp_sz - DHCP_OPTIONS_HEADER_SIZE;
+        const u_char *dhcp_option = context->buffer + dhcp_option_offset;
+        dhcp_packet_direction_t dir = (ethhdr->ether_shost[0] == context->mac[0] &&
+                                    ethhdr->ether_shost[1] == context->mac[1] &&
+                                    ethhdr->ether_shost[2] == context->mac[2] &&
+                                    ethhdr->ether_shost[3] == context->mac[3] &&
+                                    ethhdr->ether_shost[4] == context->mac[4] &&
+                                    ethhdr->ether_shost[5] == context->mac[5]) ?
+                                    DHCP_TX : DHCP_RX;
+        int offset = 0;
+        int stop_dhcp_processing = 0;
+        while ((offset < (dhcp_option_sz + 1)) && dhcp_option[offset] != 255) {
+            switch (dhcp_option[offset])
+            {
+            case 53:
```

Comment:
Better to use enum for option53, like:
enum {
  OptionDHCPMessageType = 53,
}
If we only care about option53, maybe we don;t need switch case here, just use if and remove temp variable stop_dhcp_processing.

Reply:
Updated